bundles/org.eclipse.osgi/supplement/src/org/eclipse/osgi/util/ResourceFormatter.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2012, 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - extracted from ModuleResolutionReport / ModuleRevision
+ *******************************************************************************/
+package org.eclipse.osgi.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import org.eclipse.osgi.internal.framework.FilterImpl;
+import org.osgi.framework.Constants;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.namespace.BundleNamespace;
+import org.osgi.framework.namespace.HostNamespace;
+import org.osgi.framework.namespace.PackageNamespace;
+import org.osgi.resource.Capability;
+import org.osgi.resource.Namespace;
+import org.osgi.resource.Requirement;
+
+/**
+ * Utility class that provides some generic Resource API > String conversion
+ * that are suitable to be presented to the user.
+ * 
+ * @since 3.19
+ */
+public final class ResourceFormatter {
+	public static String printCapability(Capability cap) {
```

Comment:
`print` is odd since it doesn't really print anything.  Why not `toString(Capability cap)`.

Reply:
> `print` is odd since it doesn't really print anything. Why not `toString(Capability cap)`.

I copied it from the Resource command, but of course can rename it :-)

Reply:
right, but pay much more attention to the name of public API methods that will need javadoc.  I was thinking how would we document this methods with `print` in the name when it in fact does not print anything.